### PR TITLE
fixes #5540 - also clone builtin roles during migration

### PR DESCRIPTION
**Pull request by domcleal**
Another for you @ares.  Reproducer: http://projects.theforeman.org/issues/5540

I can't see a reason _not_ to clone them, but happy to be corrected.  Without cloning builtins, we miss new roles for filters the user might have had against the builtin roles.

### CODE REVIEW

**Comment by ares** [comment]
That makes sense, anonymous role was filtered in old system so it should be cloned into new system for filtered users as well
:+1: (not tested though)

File: db/migrate/20140219183343_migrate_permissions.rb [hunk 1]
```diff
@@ -165,8 +160,7 @@ def self.migrate_user_filters
 
       say "Migrating user '#{user.login}'"
       say "... cloning all roles"
-      clones     = user.roles.builtin(false).map { |r| clone_role(r, user) }
-      user.roles = clones + user.roles.builtin(true)
+      user.roles = clones = user.roles.map { |r| clone_role(r, user) }
```
Comment (dLobatog) [inline db/migrate/20140219183343_migrate_permissions.rb:163]: I think you made a typo not removing the `= clones` from here.

Comment (domcleal) [inline db/migrate/20140219183343_migrate_permissions.rb:163]: This method uses `clones` below, so I left the variable intact.

**Comment by dLobatog** [comment]
Can confirm it works well, it created a new role Anonymous_testuser using the steps in the Redmine issue. Will merge in a second.

**Comment by dLobatog** [comment]
Merged as https://github.com/theforeman/foreman/commit/5f72ef39f3bee84942d6706e2d9954d219a56ee4 for 1.5, thanks @domcleal !

**Pull request closed without merging**